### DISCUSSION TITLE
Only partial IntelliSense mode in VSCode

### DISCUSSION THREAD

**Discussion by Splines** [motion-canvas/motion-canvas#992] [Q&A]
After creating a fresh new Motion Canvas project as described [here](https://motioncanvas.io/docs/quickstart/#creating-a-new-project) and opening the project under WSL in VSCode, I only get very limited TypeScript support. An icon in the bottom-right corner tells me:

```
Partial Mode: Project Wide IntelliSense not available
```

It also links to the [VSCode documentation](https://code.visualstudio.com/docs/nodejs/working-with-javascript#_partial-intellisense-mode) for this issue. Does anybody have already tried VSCode with Motion Canvas and has the same problem or is something with my setup flawed? Everything else in WSL works perfectly fine for me, e.g. I code Rust, Ruby on Rails, Python etc. there and all the plugins and language features are just there (after some initial setup, but that shouldn't be the case for TypeScript projects as TypeScript is first-class citizen in VSCode).

**Versions**:
- core: 3.14.1
- two: 3.14.2
- ui: 3.14.2
- vitePlugin: 3.14.1
- OS: Windows 10 -> WSL: Ubuntu 22.04.4 LTS
- Node.js: v21.5.0 (installed via nvm)
- npm: v10.5.0

**Comment by john-heinnickel** [comment]
I can't say that I've ever seen IntelliSense working in VSCode with Typescript pretty much ever....  But landed here after seeing if I might fare better with Theia's pre-built docker image, but so far no such luck.  Was worth taking a look at least.

**Comment by Splines** [comment]
In the meantime, I've opened an issue in the VSCode repo [here](https://github.com/microsoft/TypeScript/issues/60322), feel free to upvote it should you also encounter it.

**Comment by Splines** [comment]
Ok, so this was just a dumb config setting from my side that prevented TypeScript to work properly. See my answer [here](https://github.com/microsoft/TypeScript/issues/60322#issuecomment-2797813646).
